js/content/community.js

```diff
@@ -1126,51 +1126,41 @@ let GamesPageClass = (function(){
             </div>`);
     };
 
-    let scrollTimeout = null;
-
     GamesPageClass.prototype.addGamelistAchievements = function() {
         if (!SyncedStorage.get("showallachievements")) { return; }
 
         // Path of profile in view to retrieve achievement stats
         let path = window.location.pathname.replace("/games", "");
 
-        document.addEventListener("scroll", () => {
-            if (scrollTimeout) { window.clearTimeout(scrollTimeout); }
-            scrollTimeout = window.setTimeout(addAchievements, 500);
+        let observer = new IntersectionObserver(addAchievementBars, {
+            root: null, // Viewport
+            rootMargin: "0px",
+            threshold: 0.0
         });
 
-        addAchievements();
-
-        function addAchievements() {
-            let nodes = document.querySelectorAll(".gameListRow:not(.es_achievements_checked)");
-            let hadNodesInView = false;
-            for (let node of nodes) {
-
-                if (!Viewport.isElementInViewport(node)) {
-                    if (hadNodesInView) { break; }
+        async function addAchievementBars(entries) {
+            for (let entry of entries) {
+                if (!entry.isIntersecting && !(entry.boundingClientRect.top >= 0 && entry.boundingClientRect.bottom <= window.innerHeight)) {
                     continue;
                 }
 
-                hadNodesInView = true;
-
-                let appid = GameId.getAppidFromId(node.id);
-                node.classList.add("es_achievements_checked");
-                if (!node.innerHTML.match(/ico_stats\.png/)) { continue; }
```

Comment:
This check is unreliable, therefore removed. It's possible for games without this icon to have achievement stats, and vice versa.